.github/workflows/test_on_push.yml

```diff
@@ -177,6 +177,49 @@ jobs:
       - name: Upload coverage report
         uses: codecov/codecov-action@v3.1.4
 
+  test_install_odes:
+    needs: style
+    runs-on: macos-latest
+    strategy:
+      fail-fast: false
+    name: Test pybamm_install_odes on MacOS
+
+    steps:
+      - name: Check out PyBaMM repository
+        uses: actions/checkout@v4
+
+      - name: Install macOS system dependencies
+        env:
+          # Homebrew environment variables
+          HOMEBREW_NO_INSTALL_CLEANUP: 1
+          HOMEBREW_NO_AUTO_UPDATE: 1
+          HOMEBREW_NO_COLOR: 1
+          # Speed up CI
+          NONINTERACTIVE: 1
+        run: |
+          brew analytics off
+          brew update
+          brew install graphviz openblas
+
+      - name: Set up Python ${{ matrix.python-version }}
+        id: setup-python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
```

Comment:
It should pick up the Python version correctly after adding the changes requested above. The matrix did not have a Python version array earlier so it picked up the system Python 2 iiuc, which did not have `pathlib`